File: RaspberryServer/src/informations/information.cpp
#include "information.h"

Information::Information() {
}

Information::Information(std::string _author, std::string _company,
		std::string _contact, std::string _builddate,
		std::string _serverversion, std::string _websiteversion,
		std::string _temperaturelogversion, std::string _androidappversion) {
	author = _author;
	company = _company;
	contact = _contact;
	builddate = _builddate;
	serverversion = _serverversion;
	websiteversion = _websiteversion;
	temperaturelogversion = _temperaturelogversion;
	androidappversion = _androidappversion;
}

Information::~Information() {
}

std::string Information::getAuthor() {
	return author;
}

std::string Information::getCompany() {
	return company;
}

std::string Information::getContact() {
	return contact;
}

std::string Information::getBuilddate() {
	return builddate;
}

std::string Information::getServerVersion() {
	return serverversion;
}

std::string Information::getWebsiteVersion() {
	return websiteversion;
}

std::string Information::getTemperatureLogVersion() {
	return temperaturelogversion;
}

std::string Information::getAndroidAppVersion() {
	return androidappversion;
}

std::string Information::toString() {
	std::string str = std::string("Information { author: ") + author
			+ std::string("; company: ") + company + std::string("; contact: ")
			+ contact + std::string("; builddate: ") + builddate
			+ std::string("; serverversion: ") + serverversion
			+ std::string("; websiteversion: ") + websiteversion
			+ std::string("; temperaturelogversion: ") + temperaturelogversion
			+ std::string("; androidappversion: ") + androidappversion
			+ std::string(" }");
	return str;
}

File: AndroidApp/src/guepardoapps/common/classes/Information.java
package guepardoapps.common.classes;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

public class Information implements Serializable {

	private static final long serialVersionUID = 2257042096692273681L;

	private Map<String, String> _informationList;

	public Information(String author, String company, String contact, String builddate, String serverversion,
			String websiteversion, String temperaturelogversion, String androidappversion) {
		_informationList = new HashMap<String, String>();

		_informationList.put("author", author);
		_informationList.put("company", company);
		_informationList.put("contact", contact);
		_informationList.put("builddate", builddate);
		_informationList.put("serverversion", serverversion);
		_informationList.put("websiteversion", websiteversion);
		_informationList.put("temperaturelogversion", temperaturelogversion);
		_informationList.put("androidappversion", androidappversion);
	}

	public Map<String, String> GetInformationList() {
		return _informationList;
	}

	public String GetAuthor() {
		return _informationList.get("author");
	}

	public String GetCompany() {
		return _informationList.get("company");
	}

	public String GetContact() {
		return _informationList.get("contact");
	}

	public String GetBuildDate() {
		return _informationList.get("builddate");
	}

	public String GetServerVersion() {
		return _informationList.get("serverversion");
	}

	public String GetWebsiteVersion() {
		return _informationList.get("websiteversion");
	}

	public String GetTemperatureLogVersion() {
		return _informationList.get("temperaturelogversion");
	}

	public String GetAndroidAppVersion() {
		return _informationList.get("androidappversion");
	}

	public String GetKey(int index) {
		int compareIndex = 0;
		for (String key : _informationList.keySet()) {
			if (compareIndex == index) {
				return key;
			}
			compareIndex++;
		}
		return "";
	}

	public String GetValue(int index) {
		int compareIndex = 0;
		for (String key : _informationList.keySet()) {
			if (compareIndex == index) {
				return _informationList.get(key);
			}
			compareIndex++;
		}
		return "";
	}

	public String toString() {
		String information = "{Information: ";
		for (String key : _informationList.keySet()) {
			information += "{" + key + ": " + _informationList.get(key) + "};";
		}
		information += "}";
		return information;
	}
}
